prisma-fmt/src/lib.rs

```diff
@@ -65,6 +66,28 @@ pub fn lint(schema: String) -> String {
     lint::run(&schema)
 }
 
+/// Function that emulates what happened when improperly calling the `getDmmf` query engine RPC on invalid schemas.
+/// When the schema is valid, nothing happens.
+/// When the schema is invalid, the function displays a human-friendly error message indicating the schema lines
+/// where the errors lie and the total error count, e.g.:
+///
+/// ```sh
+/// The `referentialIntegrity` and `relationMode` attributes cannot be used together. Please use only `relationMode` instead.
+///   -->  schema.prisma:5
+///   |
+/// 4 |   relationMode         = "prisma"
+/// 5 |   referentialIntegrity = "foreignKeys"
+/// 6 | }
+///   |
+///
+/// Validation Error Count: 1
+/// ```
+///
+/// This function doesn't panic.
```

Comment:
We can't say that for sure, there's a lot of unwraps in parsing and validation.

Reply:
I can rephrase it as `This function isn't supposed to panic`